internal/services/containers/kubernetes_cluster_network_resource_test.go

```diff
@@ -1800,6 +1861,136 @@ resource "azurerm_kubernetes_cluster" "test" {
 `, data.RandomInteger, data.Locations.Primary, data.RandomInteger, data.RandomInteger, enablePrivateCluster, data.RandomInteger)
 }
 
+func (KubernetesClusterResource) podCidrs(data acceptance.TestData) string {
+	return fmt.Sprintf(`
+resource "azurerm_resource_group" "test" {
+  name     = "acctestRG-aks-%d"
+  location = "%s"
+}
+
+resource "azurerm_kubernetes_cluster" "test" {
+  name                = "acctestaks%d"
+  location            = azurerm_resource_group.test.location
+  resource_group_name = azurerm_resource_group.test.name
+  dns_prefix          = "acctestaks%d"
+
+  default_node_pool {
+    name       = "default"
+    node_count = 1
+    vm_size    = "Standard_DS2_v2"
+  }
+
+  network_profile {
+    network_plugin = "kubenet"
+    pod_cidrs      = ["10.1.1.0/24"]
+  }
+
+  identity {
+    type = "SystemAssigned"
+  }
+}
+`, data.RandomInteger, data.Locations.Primary, data.RandomInteger, data.RandomInteger)
+}
+
+func (KubernetesClusterResource) podCidrsDualStack(data acceptance.TestData) string {
+	return fmt.Sprintf(`
+resource "azurerm_resource_group" "test" {
+  name     = "acctestRG-aks-%d"
+  location = "%s"
+}
+
+resource "azurerm_kubernetes_cluster" "test" {
+  name                = "acctestaks%d"
+  location            = azurerm_resource_group.test.location
+  resource_group_name = azurerm_resource_group.test.name
+  dns_prefix          = "acctestaks%d"
+
+  default_node_pool {
+    name       = "default"
+    node_count = 1
+    vm_size    = "Standard_DS2_v2"
+  }
+
+  network_profile {
+    network_plugin = "kubenet"
+    pod_cidrs      = ["10.1.1.0/24", "2002::1234:abcd:ffff:c0a8:101/120"]
+    ip_versions    = ["IPv4", "IPv6"]
```

Comment:
I'm assuming this needs to be set in order to be able to provide both an IPv4 and IPv6 address? In which case we should probably add some validation in the create method to inform users as well as a note in the docs?

Reply:
Thanks! I've added that info in the doc. But I think maybe leave the complicated validation to the service side?

Reply:
The service side error message isn't particularly descriptive. AKS is complex enough as it is, so it would be beneficial to the users if we returned a more informative message such as 
`fmt.Errorf("dual-stack networking must be enabled and ip_versions must be set to [IPv4, IPv6] in order to specify multiple %s", field")`
Can you please add that in?

Reply:
Sorry I misunderstood, it's not necessary to set ` ip_versions    = ["IPv4", "IPv6"]` in order.

Reply:
I'm not sure what you mean by order but the test `TestAccKubernetesCluster_serviceCidrsDualStack` will fail if `ip_versions` isn't set to ["IPv4", "IPv6"]

Reply:
Got it, I'll test it and add this validation.

Reply:
Hi @stephybun , I just tried, it still works.
```hcl
resource "azurerm_kubernetes_cluster" "test" {
  name                = "henglu96"
  location            = azurerm_resource_group.test.location
  resource_group_name = azurerm_resource_group.test.name
  dns_prefix          = "henglu96"

  default_node_pool {
    name       = "default"
    node_count = 1
    vm_size    = "Standard_DS2_v2"
  }

  network_profile {
    network_plugin     = "kubenet"
    dns_service_ip     = "10.1.1.10"
    docker_bridge_cidr = "172.18.0.1/16"
    service_cidrs      = ["10.1.1.0/24", "2002::1234:abcd:ffff:c0a8:101/120"]
    ip_versions        = ["IPv6", "IPv4"] // Please notice here
  }

  identity {
    type = "SystemAssigned"
  }
}
```

Reply:
@ms-henglu, the order of the arguments `IPv4`/`IPv6` is irrelevant here. If a user supplies the following configuration
```
  network_profile {
    network_plugin     = "kubenet"
    dns_service_ip     = "10.1.1.10"
    docker_bridge_cidr = "172.18.0.1/16"
    service_cidrs      = ["10.1.1.0/24", "2002::1234:abcd:ffff:c0a8:101/120"]
    ip_versions        = ["IPv4"] // <---- Only one ip family
  }
```

It won't be accepted by the API
```
{
  "code": "InvalidNetworkingConfig",
  "message": "Service CIDRs do not match IP families.",
  "subcode": "",
  "target": "networkProfile.serviceCIDRs"
}
```

Reply:
Got it, I'll add validation for this case.

Reply:
@stephybun , I've added this check, thanks!